app/models/hotel.rb

```diff
@@ -1,3 +1,9 @@
 class Hotel < ApplicationRecord
   has_many :rooms
+  has_many :room_guests, through: :rooms
+  has_many :guests, through: :room_guests
+
+  def unique_guests
+    guests.distinct
```

Comment:
But either way you write your associations, this is a really good use of AR through associations!